sunxspex/emission.py

```diff
@@ -558,15 +820,99 @@ def brm2_dmlino(a, b, maxfcn, rerr, eph, eelow, eebrk, eehigh, p, q, z):
     distribution function provided in `broken_powerlaw`. The units of the computed flux is photons
     per second per keV per square centimeter.
 
     The electron flux distribution function is a double power law in electron energy with a
     low-energy cutoff and a high-energy cutoff.
 
+    Parameters
+    ----------
+    eph : np.array
+        Array of photon energies to evaluate flux at
+    p   : float
+        Slope below the break energy
+    eebrk : float
+        Break energy
+    q   : float
+        Slope above the break energy
+    eelow : float
+        Low energy electron cut off
+    eehigh : float
+        High energy electron cut off
+
+    Returns
+    -------
+    flux: np.array
+        Multiplying the output of Brm2_ThinTarget by a0 * 1.0d+55 gives an array of
+        photon fluxes in photons s^-1 keV^-1 cm^-2, corresponding to the photon energies in the input array eph.
+        The detector is assumed to be 1 AU rom the source. a0 is normalization factor in units of 1.0d55 cm-2 s-1,
+        i.e. plasma density (cm^-3) * volume of source (cm^3) * integrated electron flux density (cm^-2 s^-1)
+
+    Notes
+    -----
+    If you want to plot the derivative of the flux, or the spectral index of the photon spectrum as
+    a function of photon energy, you should set RERR to 1.d-6, because it is more sensitive to RERR
+    than the flux.
+
+    History
+    -------
+    03/15/2020, Bin Chen (binchen@njit.edu): - Initial version modified from SSW Brm2_ThinTarget.pro:
+        https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_thintarget.pro
+    """
+    # use electron flux distribution (the electron density distribution is used if EFD = False)
+    efd = True
+
+    mc2 = 510.98  # electron rest mass in keV
+    clight = 2.9979e10  # speed of light cm/s
+    au = 1.496e13  # astronomical unit cm
+    r0 = 2.8179e-13  # classical electron radius cm
+
+    # Max number of points
+    maxfcn = 2048
+
+    # Average atomic number
+    z = 1.2
+
+    # Relative error
+    rerr = 1e-4
+
+    # Numerical coefficient for photo flux
+    fcoeff = (clight / (4 * np.pi * au ** 2)) / mc2 ** 2.
+
+    # Create arrays for the photon flux and error flags.
+    flux = np.zeros_like(eph, dtype=np.float64)
+    iergq = np.zeros_like(eph, dtype=np.float64)
+
+    if eelow >= eehigh:
+        raise ValueError('eehigh must be larger than eelow!')
+        return flux
+
+    l, = np.where((eph < eehigh) & (eph > 0))
+    if l.size > 0:
+        #pdb.set_trace()
+        flux[l], iergq[l] = brm2_dmlin(eph[l], np.full_like(l, eehigh), maxfcn, rerr,
+                                       eph[l], eelow,
+                                       eebrk, eehigh, p, q, z, efd)
+
+        flux *= fcoeff
+
+        return flux
+    else:
+        raise Warning('The photon energies are higher than the highest electron energy or not greater than zero')
+
+def Brm2_ThickTarget(eph, p, eebrk, q, eelow, eehigh):
```

Comment:
Trying to make the names more descriptive perhaps `bremsstrahlung_thick_target` also python function usually use snake case.

Reply:
done